plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -441,6 +477,72 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
+    public static ColumnMapping timestampColumnMapping(TimestampType timestampType)
+    {
+        if (timestampType.getPrecision() <= MAX_SHORT_PRECISION) {
+            // SQLServer JDBC driver does not support LocalDateTime so timestampColumnMapping() cannot be used here
+            return ColumnMapping.longMapping(
+                    timestampType,
+                    shortTimestampReadFunction(),
+                    shortTimestampWriteFunction());
+        }
+        return ColumnMapping.objectMapping(
+                timestampType,
+                longTimestampReadFunction(),
+                longTimestampWriteFunction());
+    }
+
+    private static LongReadFunction shortTimestampReadFunction()
+    {
+        return (resultSet, columnIndex) -> {
+            Timestamp timestamp = resultSet.getTimestamp(columnIndex, newUtcCalendar());
```

Comment:
did you try  `.getString`?

does the driver support `getObject(, Class)`? if it doesn, does it have any non-std representation for date/time?

Reply:
I tried `getObject(..., LocalDateTime.clas)` and it failed. Haven't tried with a string.

Reply:
Please try. Working with string makes it much nicer most of the time, than struggling with ancient java.sql API.

Reply:
I checked wrong earlier and it turns out LocalDateTime **is** supported. It didn't occur to me to check which version of mssql-jdbc we're using and it's pretty old. I bumped it to the next stable version which is 7.2.1 (it's not the latest).

It looks like they do the conversion the same way I tried. But I think it is better to push this responsibility down into the driver.

https://github.com/microsoft/mssql-jdbc/pull/749

Reply:
Bumped to the latest 8.4.1 after talking offline with @losipiuk

Reply:
Extract separate commit with bump please.